build/Targets/Settings.props

```diff
@@ -81,6 +81,16 @@
     <ProjectAssetsFile Condition="'$(BuildingForLiveUnitTesting)' == 'true'">$(LiveUnitTestingOriginalBaseIntermediateOutputPath)project.assets.json</ProjectAssetsFile>
   </PropertyGroup>
 
+  <!-- Utilize source linking whenever there is a valid git commit available. Won't be available during rebase, 
+       or build from zip for example -->
+  <PropertyGroup Condition="'$(GitHeadShaRaw)' != '' AND '$(OS)' == 'Windows_NT' ">
+    <SourceLinkCreate>true</SourceLinkCreate>
+    <SourceLinkRootDirectory Condition="'$(SourceLinkRootDirectory)' == ''">$(RepoRoot)</SourceLinkRootDirectory>
+    <SourceLinkOriginUrl>https://github.com/dotnet/roslyn</SourceLinkOriginUrl>
+    <SourceLinkUrl>https://raw.githubusercontent.com/dotnet/roslyn/$(GitHeadShaRaw)/*</SourceLinkUrl>
+    <SourceLinkCommit>$(GitHeadShaRaw)</SourceLinkCommit>
```

Comment:
@ctaggart, @tmat 

> If SourceLinkUrl and the SourceLinkCommit are not set, they are figured out by running git commands.

I noticed that when looking at several of your samples / implementation. The reason for still using the explicit values is build performance and correctness.

- Performance: Letting SourceLink determine the values causes 6 processes to be spawned per project built: the 3 that SourceLink will spawn to calculate root, sha and URL. Each of these is wrapped in an `cmd.exe` process that comes with using `<Exec>`. There are ~170 projects in Roslyn so that would mean about ~1,000 processes spawned during a build. That causes a 15-20% increase in build time for us as compared with just using the values we already know.
- Correctness: The calculation of `SourceLinkUrl` can produce different values depending on how the repo was cloned. For instance on my box it uses my fork vs. the official repo. Prefer to be explicit here, in particular for official builds, and make sure we are setting up the correct server info. We've gotten burned in the past making assumptions about how official builds are cloned. 

For 99% of the projects out there I think the approach SourceLink takes will work great. Roslyn is just a huge project that needs to tolerate being built in a number of different ways with different environments. 

> SourceLinkOriginUrl isn't needed because you are setting SourceLinkUrl

Plan is to eventually use SourceLinkUrl. Presently using SourceLinkUrl will cause SourceLink to log a number of high priority messages to the build output. Using SourceLinkOriginUrl temporarily to avoid that.

I still owe you a PR to get the priority of those messages changed. Haven't forgotten about it, just gotten bogged down in a number of issues around shipping VS2017 Update 3. 😦 

> Looks great!

Thanks for taking the time to look at this 😄